UnitystationLauncher/Views/InstallationView.axaml.cs

```diff
@@ -14,4 +18,37 @@ private void InitializeComponent()
     {
         AvaloniaXamlLoader.Load(this);
     }
+    
+    private void HandlePathClick(object sender, PointerPressedEventArgs e)
+    {
+        TextBlock? block = sender as TextBlock;
+        if (block?.Text == null) return;
+        string path = Path.GetFullPath(block.Text);
+        string protcol = GetManagerProtocl();
+        Process.Start(protcol, new Uri(RemoveLastPart(path)).ToString());
+    }
+
+    private string GetManagerProtocl()
+    {
+        string protocol = "Explorer.exe";
+        if (OperatingSystem.IsWindows())
```

Comment:
We have the EnvironmentService to do platform detection already, if you move is from the View to the ViewModel using a Command instead of a PointerClicked event it should be as easy as dependency injecting it in.

Reply:
Actually, I don't think you need to detect the platform at all, you can just do it the same way we open log directory after a scan fail, it should open in whatever the user has specified as their default handler for opening directories.

https://github.com/unitystation/stationhub/blob/develop/UnitystationLauncher/Services/InstallationService.cs#L550

```csharp
ProcessStartInfo psi = new()
{
    FileName = path,
    UseShellExecute = true,
    Verb = "open"
};
```

Reply:
This doesn't work.
```
Unhandled exception. System.ComponentModel.Win32Exception (5): An error occurred trying to start process 'C:\Users\Maximus\AppData\Roaming\StationHub\Installations\UnityStationDevelop' with working directory 'F:\projects\stationhub\UnitystationLauncher\bin\Debug\net8.0'. Access is denied.
```

Reply:
> We have the EnvironmentService to do platform detection already, if you move this from the View to the ViewModel using a Command instead of a PointerClicked event it should be as easy as dependency injecting it in.

what difference does this make?

Reply:
> This doesn't work.
> 
> ```
> Unhandled exception. System.ComponentModel.Win32Exception (5): An error occurred trying to start process 'C:\Users\Maximus\AppData\Roaming\StationHub\Installations\UnityStationDevelop' with working directory 'F:\projects\stationhub\UnitystationLauncher\bin\Debug\net8.0'. Access is denied.
> ```

Hmm, thats odd, I checked it out locally and that worked fine for me. Does it work when opening the log folder for a failed scan? I tested that in a Windows VM when I was working on that before and I remember it working fine, but maybe something regarding that changed with the upgrade to .NET 8.

> > We have the EnvironmentService to do platform detection already, if you move this from the View to the ViewModel using a Command instead of a PointerClicked event it should be as easy as dependency injecting it in.
> 
> what difference does this make?

Having standard ways of doing things makes the code a lot more maintainable, and code in the ViewModel can be effectively unit tested, while the code in the View cannot be in most cases. https://docs.avaloniaui.net/docs/concepts/the-mvvm-pattern